### PR TITLE
feat(atomic): improve recs interface initialization to be on par with search

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-2074

### CODE REVIEW

File: packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx [hunk 1]
```diff
@@ -43,13 +49,36 @@ export class AtomicRecsInterface
...
+
   /**
    * The recommendation interface headless engine.
    */
   @Prop({mutable: true}) public engine?: RecommendationEngine;
+
+  /**
+   * The search interface [query pipeline](https://docs.coveo.com/en/180/).
+   */
+  @Prop({reflect: true}) public pipeline?: string;
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx:80]: From the docs https://docs.coveo.com/en/l25f0432/leverage-machine-learning/associate-a-content-recommendations-cr-model-with-a-query-pipeline it seems to be relevant ? We can add a note to use the recommendation property instead?

File: packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx [hunk 2]
```diff
@@ -43,13 +49,36 @@ export class AtomicRecsInterface
...
+   * The recommendation interface [search hub](https://docs.coveo.com/en/1342/).
+   */
+  @Prop({reflect: true}) public searchHub = 'default';
+
+  /**
+   * The [tz database](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) identifier of the time zone to use to correctly interpret dates in the query expression, facets, and result items.
+   * By default, the timezone will be [guessed](https://day.js.org/docs/en/timezone/guessing-user-timezone).
+   *
+   * @example
+   * America/Montreal
```
Comment (jpmarceau) [inline packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx:68]:
```suggestion
   * Example: "America/Montreal"
```

**Review by jpmarceau — APPROVED** [review]
👍

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx (hunk 1 above)
Comment (olamothe) [reply packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.tsx:79]: Yeah the recommended (hah) way is to set a search hub + recommendation ID.

And the the query pipeline condition becomes `if searchhub=X and recommendation=Recommendation`

**Review by btaillon — APPROVED** [review]

**Pull request merged**